pkg/operators/cli/clioperator.go

```diff
@@ -205,7 +205,26 @@ func (o *cliOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error
 			fields, hasFields = fieldLookup[""] // fall back to default
 		}
 
-		switch o.mode {
+		mode := o.mode
+		if annotatedMode := ds.Annotations()["cli.output"]; annotatedMode != "" {
+			mode = annotatedMode
+		}
```

Comment:
Agree, it should be overridable by the user. Will see how it fits in.

Regarding supported output modes: I think that should be determined by the CLI operator itself - if it should be restricted at all. I initially had support for setting defaults for params in gadget.yaml in this PR, but decided to drop it (and reintroduce it later on its own), because it is very similar to the output mode annotation for this scenario.

Reply:
Also because we need the `-o json` to run the tests with the ig framework.